### PR TITLE
[CDAP-12432] Adds description and delete actions to individual rules in rules engine

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Adds description to rules
- Adds delete action to delete rules
- Fixes long rule and rulebook names

JIRA:
- https://issues.cask.co/browse/CDAP-12414
- https://issues.cask.co/browse/CDAP-12432

### CODE REVIEW

File: cdap-ui/app/cdap/components/RulesEngineHome/RulesTab/index.js [hunk 1]
```diff
@@ -120,9 +120,12 @@ export default class RulesTab extends Component {
             null
         }
         {
-          rules.map(rule => {
+          rules.map((rule, i) => {
             return (
-              <Rule rule={rule} />
+              <Rule
+                rule={rule}
+                key={i}
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/RulesEngineHome/RulesTab/index.js]: is the order of the array guaranteed to be consistent? for example if I add another rule, will it get added as the last item of the array? If not, setting the index as key might cause a problem. Might be safer to use `rule.id` (if it exists)

File: cdap-ui/app/cdap/components/RulesEngineHome/RulesTab/Rule/Rule.scss [hunk 2]
```diff
@@ -32,6 +32,15 @@ $drag_bars_color: gray;
       padding-left: 0;
     }
 
+    .rule-name-container {
+      display: flex;
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/RulesEngineHome/RulesTab/Rule/Rule.scss:36]: if there is only one child for this container, do you need to set it to display flex?

Comment (ajainarayanan) [reply cdap-ui/app/cdap/components/RulesEngineHome/RulesTab/Rule/Rule.scss:36]: Its not single element. It has svg arrow and the name.

**Review by elfenheart — APPROVED** [review]
👍  LGTM

**Review by bdmogal — APPROVED** [review]

**Pull request merged**